lib/active_merchant/billing/gateways/blue_snap.rb

```diff
@@ -531,6 +550,8 @@ def root_attributes
       end
 
       def root_element(action, payment_method_details)
+        return 'refund' if action == :refund
```

Comment:
I like how you've done this 👍 

I think the method could be even better if we take your idea further:
```
def root_element(action, payment_method_details)
  return 'refund' if action == :refund
  return 'vaulted-shopper' if action == :store

  payment_method_details.root_element
end
```

Reply:
Looks better, changes are updated.